File: src/bin/02-2/main.py
with open('data/02-1/input.txt', 'r') as input_file:
    input = input_file.read()


input = input.split('\n')
good_games = []

for line in input:
    if len(line) == 0:
        continue

    game_id = int(line[4:].split(':')[0])
    bad_game = False
    sets = line.split(':')[1].split(';')
    sets = [item.strip() for item in sets]

    min_set = [0,0,0] # rgb
    for set_i, set in enumerate(sets):
        colors = set.split(',')
        for item in colors:
            duo = item.strip().split(' ')
            qty, color = duo[0], duo[1]
            if color == "red":
                if int(qty) > min_set[0]:
                    min_set[0] = int(qty)
            elif color == "green":
                if int(qty) > min_set[1]:
                    min_set[1] = int(qty)
            elif color == "blue":
                if int(qty) > min_set[2]:
                    min_set[2] = int(qty)

    power = min_set[0] * min_set[1] * min_set[2]
    good_games.append(power)

print(sum(good_games))

File: src/bin/02-2/main.rs
use std::path::Path;


fn main() {
    let input = match std::fs::read_to_string(Path::new("./data/02-1/input.txt")) {
        Ok(input_str) => input_str,
        Err(error) => {
            println!("Error while opening input: {}", error);
            return;
        }
    };

    let mut sum: u32 = 0;

    for (_line_index, line) in input.lines().enumerate() {
        let first: Vec<&str> = line.split(':').collect();
        let sets: Vec<&str> = first[1].trim().split(';').collect();
        let mut min: [u32; 3] = [0, 0, 0];

        for set in sets.iter() {
            let color_data: Vec<&str> = set.split(',').collect();
            for data in color_data.iter() {
                let tmp: Vec<&str> = data.trim().split(' ').collect();
                let qty = tmp[0].parse::<u32>().unwrap();
                let color = tmp[1];
                if color == "red" {
                    if qty > min[0] {
                        min[0] = qty;
                    }
                } else if color == "green" {
                    if qty > min[1] {
                        min[1] = qty;
                    }
                } else if color == "blue" {
                    if qty > min[2] {
                        min[2] = qty;
                    }
                }
            }
        }
        
        sum += min[0] * min[1] * min[2];
    }
    println!("{sum}");
}
